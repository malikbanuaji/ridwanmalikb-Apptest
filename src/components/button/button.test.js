import * as React from 'react';
import Button from './button';

import {render} from '@testing-library/react-native';

test('should render properly', () => {
  const component = render(<Button title="Test" />);
  expect(component.toJSON()).toMatchSnapshot();
});

test('should render properly with danger type', () => {
  const component = render(<Button title="Test" type="danger" />);
  expect(component.toJSON()).toMatchSnapshot();
});

test('should render properly with disabled', () => {
  const component = render(
    <Button type="no_type" title="Test" disabled={true} />,
  );
  expect(component.toJSON()).toMatchSnapshot();
});
