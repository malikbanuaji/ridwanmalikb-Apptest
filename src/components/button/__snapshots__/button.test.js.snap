// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 4,
        "overflow": "hidden",
      },
      Object {
        "backgroundColor": "#1e90ff",
      },
    ]
  }
>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "paddingHorizontal": 10,
          "paddingVertical": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 16,
              "fontWeight": "800",
            },
            Object {
              "color": "#ffffff",
            },
          ]
        }
      >
        Test
      </Text>
    </View>
  </View>
</View>
`;

exports[`should render properly with danger type 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 4,
        "overflow": "hidden",
      },
      Object {},
    ]
  }
>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "paddingHorizontal": 10,
          "paddingVertical": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 16,
              "fontWeight": "800",
            },
            Object {
              "color": "#f57731",
            },
          ]
        }
      >
        Test
      </Text>
    </View>
  </View>
</View>
`;

exports[`should render properly with disabled 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 4,
        "overflow": "hidden",
      },
      Object {
        "opacity": 0.5,
      },
      Object {
        "backgroundColor": "#1e90ff",
      },
    ]
  }
>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "paddingHorizontal": 10,
          "paddingVertical": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 16,
              "fontWeight": "800",
            },
            Object {
              "color": "#f57731",
            },
          ]
        }
      >
        Test
      </Text>
    </View>
  </View>
</View>
`;
